product/src/test/java/org/itsallcode/openfasttrace/cli/ProcessOutputConsumer.java

```diff
@@ -0,0 +1,147 @@
+package org.itsallcode.openfasttrace.cli;
+
+import java.io.*;
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.concurrent.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+class ProcessOutputConsumer
+{
+    private static final Logger LOG = Logger.getLogger(ProcessOutputConsumer.class.getName());
+    private final Executor executor;
+    private final Process process;
+    private final ProcessStreamConsumer stdOutConsumer;
+    private final ProcessStreamConsumer stdErrConsumer;
+
+    ProcessOutputConsumer(final Process process)
+    {
+        this(createThreadExecutor(), process, new ProcessStreamConsumer("stdout"), new ProcessStreamConsumer("stderr"));
+    }
+
+    ProcessOutputConsumer(final Executor executor, final Process process,
+            final ProcessStreamConsumer stdOutConsumer, final ProcessStreamConsumer stdErrConsumer)
+    {
+        this.executor = executor;
+        this.process = process;
+        this.stdOutConsumer = stdOutConsumer;
+        this.stdErrConsumer = stdErrConsumer;
+    }
+
+    private static Executor createThreadExecutor()
+    {
+        return runnable -> {
+            final Thread thread = new Thread(runnable);
+            thread.setName(ProcessOutputConsumer.class.getSimpleName());
```

Comment:
Wouldn't it make more sense to name the thread after the process being consumed? Or at least to get that processes identification into the name? Thread names are mainly a debugging means as far as I remember.